packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -142,14 +148,17 @@ export function getRootHostContext(
   if (__DEV__) {
     const validatedTag = type.toLowerCase();
     const ancestorInfo = updatedAncestorInfo(null, validatedTag);
+    if (enableEventAPI) {
+      return {namespace, ancestorInfo, isEventComponent: false};
+    }
     return {namespace, ancestorInfo};
   }
   return namespace;
 }
 
 export function getChildHostContext(
```

Comment:
It would be better to have a `getChildHostContextForEvent` function in the HostConfig. That way there can be a default noop version that just returns the parent context that will be inlined into beginWork.

We also don't have to consider the complexities of events when dealing with just simple host components.